lib/si-layer-cache/src/db/rebase_batch.rs

```diff
@@ -102,7 +102,7 @@ where
         let mut tried = 0;
         let read_wait = Instant::now();
         while tried < MAX_TRIES {
-            if let Some(v) = self.cache.memory_cache().get(&key).await {
+            if let Some(v) = self.cache.cache().get(&key).await {
```

Comment:
My first thought was that Foyer appears to check in memory (and early return) and then fall back to check on disk, and so our retry loop may no longer apply. However, does the `self.cache.get(..).await` call below fall back to querying postgres currently? If so, then I understand 😄

Reply:
This one calls directly into the cache, so no postgres. I _think_ that we're waiting for the rebase to come across nats and be populated into memory here, hence the loop.

Reply:
Right, that does make sense!